src/t_zset.c

```diff
@@ -3760,16 +3766,36 @@ void genericZpopCommand(client *c, robj **keyv, int keyc, int where, int emitkey
         return;
```

Comment:
we'll need to add some flag to this function to control if this reply is an empty array a NIL, right?

Reply:
Yes, agree with the NIL response, can we just check `if (c->cmd->proc == zmpopCommand) {` and return a NIL?

Reply:
i think i'd rather that to be an explicit input flag passed from outside, and not a hackish check on the client / command.
we already have another flag specifying the response format.
e.g. the boolean flag can be `reply_nil_when_empty` or such.

maybe an alternative to that is check `use_nested_array && emitkey` with some comment saying:
```c
/* In ZMPOP / BZMPOP we reply with a structured nested array containing key name
 * and member + score pairs. in these commands, we reply with null when we have no result */
if (use_nested_array && emitkey) {
    addReply(c, shared.null[c->resp]);
} else {
    addReply(c, shared.emptyarray);
}
```
so this becomes a logical decision, rather than something specifically set per command.

Reply:
I did think about a extra input flag and `use_nested_array && emitkey`, and eventually camp up with `c->cmd->proc == zmpopCommand`.

After a rethinking, my origin thought was to minimize the changes, and yes, hackish check indeed a bad way
`use_nested_array && emitkey` it is indeed can do the work and with less changes. PS: BZMPOP won't follow this path (we will check the llen before we enter `genericZpopCommand`)

At last, i thought about it and decided to use `reply_nil_when_empty`, we also have `use_nested_array` flag, so i think it is the right way, and with some comments, the code is also easy to follow and maintain

Reply:
I expected to see an update to the tests in the last commit. 
Maybe it's not detected and we need to use `readraw` mode.